test/logflare/sources_test.exs

```diff
@@ -82,4 +81,28 @@ defmodule Logflare.SourcesTest do
       assert left_schema == schema
     end
   end
+
+  describe "preload_for_dashboard/1" do
+    setup do
+      %{user: insert(:user)}
+    end
+
+    test "preloads required fields", %{user: user} do
+      sources = insert_list(3, :source, %{user: user})
+      sources = Sources.preload_for_dashboard(sources)
+
+      assert Enum.all?(sources, &Ecto.assoc_loaded?(&1.user))
+      assert Enum.all?(sources, &Ecto.assoc_loaded?(&1.rules))
+      assert Enum.all?(sources, &Ecto.assoc_loaded?(&1.saved_searches))
+    end
+
+    test "sorts data by name and favorite flag", %{user: user} do
+      source_1 = insert(:source, %{user: user, name: "C"})
```

Comment:
lets use lists for the factory attrs, less verbose

Reply:
👍